src/AutomatedCar/Models/ExternalGearbox.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace AutomatedCar.Models
+{
+    using Avalonia.Media;
+    using ReactiveUI;
+    using System;
+    using SystemComponents;
+
+    public class ExternalGearbox : ReactiveObject
+    {
+        public enum Gear { P = 0, R = 1, N = 2, D = 3 }
+        private Gear currentExternalGear = Gear.P;
+        private AutomatedCar automatedCar;
+
+        public ExternalGearbox(AutomatedCar automatedCar)
+        {
+            this.automatedCar = automatedCar;
+        }
+
+        public Gear CurrentExternalGear
+        {
+            get
+            {
+                return this.currentExternalGear;
+            }
+            set
+            {
+                this.RaiseAndSetIfChanged(ref this.currentExternalGear, value);
+            }
+        }
+
+        public void ExternalUpshift()
```

Comment:
The method name can be simply Upshift which inherited from the interface.

Reply:
I renamed it, thank you.